rpcs3/Emu/RSX/VK/VKGSRender.cpp

```diff
@@ -392,6 +392,25 @@ VKGSRender::VKGSRender() : GSRender()
 	m_device = const_cast<vk::render_device*>(&m_swapchain->get_device());
 	vk::set_current_renderer(m_swapchain->get_device());
 
+	// Device-specific overrides
+	if (g_cfg.video.vk.asynchronous_texture_streaming)
+	{
+		if (m_device->get_graphics_queue() == m_device->get_transfer_queue())
+		{
+			rsx_log.error("Cannot run graphics and async transfer in the same queue. Async uploads are disabled. This is a limitation of your GPU");
+			g_cfg.video.vk.asynchronous_texture_streaming.set(false);
```

Comment:
since this is a dynamic option, it seems like a hazard to do it here.

Reply:
You can also set a boolean override for it like m_aync_texture_stream_allowed and check it as well everytime you read the setting.

Reply:
This option affects things outside the RSX thread itself and would require rsx::thread::backend_config to be global state. Since I don't want to bother touching fxo with it right now I'll just leave it as is (requires a significant restructure of how options are read). Dynamic toggle has been removed from this option.